core/src/main/java/org/apache/calcite/sql/type/SqlTypeName.java

```diff
@@ -65,6 +65,8 @@
   DATE(PrecScale.NO_NO, false, Types.DATE, SqlTypeFamily.DATE),
   TIME(PrecScale.NO_NO | PrecScale.YES_NO, false, Types.TIME,
       SqlTypeFamily.TIME),
+  DATETIME(PrecScale.NO_NO | PrecScale.YES_NO, false, Types.OTHER,
```

Comment:
Why we have added DATETIME here ?

Reply:
As QA team requirements, we need to convert DATETIME to DATETIME (Earlier DATETIME TO DATE in existing code), in **DataTypeMapping**(Swift) earlier mapped with DATE because DATETIME was unavailable in SqlTypeName.

Reply:
@vinayujeedm map it with a standard datatype which is already present in SqlTypeName, and then handle target specific changes in target dialect. You can refer to BigQuerySqlDialect class where it is already implemented.